### DISCUSSION TITLE
Why is the output different in `Windows Terminal` and vscode terminal??

### DISCUSSION THREAD

**Discussion by Dagwbl** [quarto-dev/quarto-cli#2647] [Q&A]
This is the output of the `Windows Termianl`:
```bash
❯ quarto check

[>] Checking Quarto installation......OK
      Version: 1.2.160
      Path: D:\Applications\Scoop\apps\quarto-prerelease\current\bin
      CodePage: 936

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: D:/ProgramData/Miniconda3/python.exe
      Jupyter: 4.11.1
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.2.1
      Path: D:/Applications/winget/r
      LibPaths:
        - D:/Applications/winget/r/library
      rmarkdown: 2.16.1

[>] Checking Knitr engine render......OK
```
And this is VSCode termianl
```bash
❯ quarto check jupyter

WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: D:/ProgramData/Miniconda3/python.exe
      Jupyter: 4.11.1
      Kernels: python3

(-) Checking Jupyter engine render....WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Jupyter engine render....OK

 Sky@Dagwbl  E:  Documents  Rmarkdown  Thesis  main   base 3.10.6 
❯ quarto check

[>] Checking Quarto installation......OK
      Version: 1.2.160
      Path: D:\Applications\Scoop\apps\quarto-prerelease\current\bin
      CodePage: 936

[>] Checking basic markdown render....OK

WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: D:/ProgramData/Miniconda3/python.exe
      Jupyter: 4.11.1
      Kernels: python3

(/) Checking Jupyter engine render....WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.2.1
      Path: D:/Applications/winget/r
      LibPaths:
        - D:/Applications/winget/r/library
      rmarkdown: 2.16.1

[>] Checking Knitr engine render......OK
```

In addition, a warning was thrown on vscode termianl. `D:\ProgramData\Anaconda3\envs\python39\python.exe` It really doesn't exist.

**Comment by mcanouil** [comment]
What are you calling Windows terminal? PowerShell (which version)? DOS terminal? Something else?
What's your default terminal in VSCode on Windows?

Also, you did not use the same command for both outputs, or more accurately you executed two commands in one.
The main difference seems to be related to environment variable which does not seem to be discovered properly.

**Reply by Dagwbl** [reply]
Both of them call the default powershell

**Comment by Dagwbl** [comment]
> What are you calling Windows terminal? PowerShell (which version)? DOS terminal? Something else?
> What's your default terminal in VSCode on Windows?
> 
> Also, you did not use the same command for both outputs, or more accurately you executed two commands in one.
> The main difference seems to be related to environment variable which does not seem to be discovered properly.

https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701?hl=zh-cn&gl=cn

That's what I mean windows termianl.


I'm sorry I can't view the powershell version right now.
And I think it may be that vscode appends other environment variables

**Reply by Dagwbl** [reply]
Even if I use quarto check, a warning will be thrown. Now I leave my computer, and I will add additional information in ten hours

**Reply by mcanouil** [reply]
What if you use PowerShell terminal, do you have the same warning? (to get PowerShell version you can type ` $PSVersionTable.PSVersion`).

For your information, the native/integrated Windows PowerShell version is 5.1, while there is 7.2 available (https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2).

**Reply by Dagwbl** [reply]
Ok, I will try tomorrow. Thanks for the reply.

**Reply by Dagwbl** [reply]
> What if you use PowerShell terminal, do you have the same warning? (to get PowerShell version you can type ` $PSVersionTable.PSVersion`).
> 
> For your information, the native/integrated Windows PowerShell version is 5.1, while there is 7.2 available (https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2).

I got the same results in three clients(native powershell, vscode terminal, windows terminal):
```bash
❯ $PSVersionTable.PSVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      22621  607
```

Only when there is a warning on vscode terminal
```bash
❯ quarto check

[>] Checking Quarto installation......OK
      Version: 1.2.161
      Path: D:\Applications\Scoop\apps\quarto-prerelease\current\bin
      CodePage: 936

[>] Checking basic markdown render....OK

WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: D:/ProgramData/Miniconda3/python.exe
      Jupyter: 4.11.1
      Kernels: python3

(/) Checking Jupyter engine render....WARNING: Specified QUARTO_PYTHON 'D:\ProgramData\Anaconda3\envs\python39\python.exe' does not exist.
[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.2.1
      Path: D:/Applications/winget/r
      LibPaths:
        - D:/Applications/winget/r/library
      rmarkdown: 2.16.1

[>] Checking Knitr engine render......OK
```

**Comment by Dagwbl** [comment]
I change the default terminal to `PowerShell 7.2`, also `quarto check` didn't throw the warning.
but cell chunk didn't run successfully and not thrown any error when I clicked `run cell`:
![image](https://user-images.githubusercontent.com/39078825/192925366-53ee70df-2dc0-4196-a88a-0b4a5bf0fca7.png)

**Comment by Dagwbl** [comment]
Then I installed the native python from https://www.python.org/, I clicked the `run cell` button will toggle a dialog:

![image](https://user-images.githubusercontent.com/39078825/192926913-2f1e37e4-4f93-4b8c-b152-0c70ce271b8c.png)

Then clicked the `change kernel` button, the cell can run successfully. Confusing...

**Reply by mcanouil** [reply]
Hum, the issue is that I can't really reproduce your setup and thus the warning.
Maybe someone from Quarto team have some insights about how the `QUARTO_PYTHON` is set, thus what might be the exact reason behind.

**Reply by Dagwbl** [reply]
`QUARTO_PYTHON` seems that it does not seem to work on Windows. 
It should be the extension of vscode to create residual files. I think reinstalling vscode will solve the problem. I usually use knitr, so I'll make do with it.

**Reply by mcanouil** [reply]
I am not sure `QUARTO_PYTHON` is something created by the Quarto for VSCode extension.

To note, you could define `QUARTO_PYTHON` (or any environment variables) on Windows in the user/admin settings of your computer.

**Reply by Dagwbl** [reply]
Maybe Jupyter extension or python extension, I have defined `QUARTO_PYTHON` on system variables, but not defined on user variables.
I try it now.

**Comment by cderv** [comment]
@Dagwbl did you find where you could have set `QUARTO_PYTHON` ? 

It seems it is defined somewhere and picked up by the VSCODE extension. I would checked the different places where you have set the variable. We can then see why it is trhowing a warning only on VSCODE.

**Reply by Dagwbl** [reply]
![image](https://user-images.githubusercontent.com/39078825/193036940-c4d1245e-91a1-4246-8d7f-550463406de6.png)

The user variables is what I just set but I haven't restarted my computer yet. The system variable was set when I installed quarto before.

**Reply by Dagwbl** [reply]
It looks like the same problem as before, but now I upgrade pwershell and there is no warning message.

**Reply by Dagwbl** [reply]
I'm sorry I did a lot of things to interfere with checking bug. 
Now I can't reproduce the warning
Now my quarto works almost normally, except that it always chooses the default native python, I am uninstalling the native python to check the problem.

**Reply by Dagwbl** [reply]
After I uninstalled the native python, there was no response when I clicked `run cell`

**Reply by cderv** [reply]
I believe Quarto VSCODE extension leverage the Python extension for using python. So you should check that this extension is correctly working and that VSCODE finds a python environment to run

**Reply by mcanouil** [reply]
What do yo call "native Python"? From the environment variables (and before), you are not using A standard Python install, but instead some conda environment from which you get Python.

I would suggest to uninstall/remove Miniconda3 and Anaconda3. To uninstall/remove Quarto. To remove any environment variables related to any of those tools.
Then, install the distribution/environment from which you want to use Python.
Test in PowerShell terminal from Windows and in PowerShell via VSCode your Python environment.
Finally reinstall the latest stable (or pre-release) version of Quarto and check first PowerShell terminal from Windows, then from within VSCode.

**Reply by Dagwbl** [reply]
Yeap! It's work successfully after I set the default interpreter path for python extension.

About the doesn't nonexistent path: `D:\ProgramData\Anaconda3\envs\python39\python.exe`. I'm almost sure it was the configuration file generated by the vscode extension when I used anaconda before (such as jupyter extension or python extension). Today I also uninstalled and reinstalled the jupyter and python extensions.
> I believe Quarto VSCODE extension leverage the Python extension for using python. So you should check that this extension is correctly working and that VSCODE finds a python environment to run

Thank you for giving me this crucial information.

**Reply by Dagwbl** [reply]
It's a terrible that change hastily the development environment. I will pay attention in the future.
Thank you! @mcanouil @cderv

**Reply by mcanouil** [reply]
When using VSCode, you can check the settings that has been modified using `@modified` in the settings search bar, you can add "Python" to search for anything.
This is a useful tip when trying to figure out why there is some behaviour in VSCode.

**Reply by Dagwbl** [reply]
> When using VSCode, you can check the settings that has been modified using `@modified` in the settings search bar, you can add "Python" to search for anything. This is a useful tip when trying to figure out why there is some behaviour in VSCode.

So cool! This is a really useful skill.
